main.go

```diff
@@ -726,7 +726,16 @@ func parseFlags() (config, error) {
 		rawTracingEndpointType  string
 	)
 
-	cfg := config{}
+	// if at least one metrics/logs endpoint (read or write) is set
+	// metrics/logs will be enabled.
+	cfg := config{
+		metrics: metricsConfig{
+			enabled: false,
+		},
+		logs: logsConfig{
+			enabled: false,
+		},
+	}
```

Comment:
Why is that manual defaulting needed? Golang's type system takes care of this.

Reply:
Totally right, closing it.